In [213]:
import pandas as pd

In [214]:
file='commentsRedditVaccines.csv'
df = pd.read_csv(file, encoding='utf-8')
print('Length of dataset is: ',len(df))
print('The columns are: \n',df.columns)

Length of dataset is:  34768
The columns are: 
 Index(['post_id', 'post_author', 'post_date', 'post_title', 'post_score',
       'post_permalink', 'post_url', 'comment_id', 'comment_author',
       'comment_date', 'comment_parent_id', 'comment_edited', 'comment_score',
       'comment_body'],
      dtype='object')


In [215]:
row_headers=['post_id','post_title','post_score','comment_id','comment_body']
df = pd.read_csv(file, names=row_headers,skiprows=1)
print(df.iloc[0:2])

                                                                                                                                                                                                                                          post_id  \
kdmkbz KinnerNevada 2020-12-15 14:21:20 Moderna's vaccine is highly effective, FDA says... 42481 /r/Coronavirus/comments/kdmkbz/modernas_vaccine... https://www.nbcnews.com/health/health-news/mode... gfx8br4 AutoModerator  2020-12-15 14:21:21   
                                                                                                                                                                                                       gfx930g Hothabanero6   2020-12-15 14:28:43   

                                                                                                                                                                                                                             post_title  \
kdmkbz KinnerNevada 2020-12-1

In [216]:
print (df.columns)

Index(['post_id', 'post_title', 'post_score', 'comment_id', 'comment_body'], dtype='object')


In [217]:
df.shape

(34768, 5)

In [218]:
[print("Unique",header,':',df[header].nunique()) for header in row_headers]

Unique post_id : 34481
Unique post_title : 17438
Unique post_score : 1559
Unique comment_id : 576
Unique comment_body : 32239


[None, None, None, None, None]

There are 34481 unique posts found in this data.

17438 unique titles.

1559 unique scores or karma points since this dataset is from reddit discussion boards.

576 unique comments are found in the dataset but despite that there are 32239 unique comments or comment_body written in the dataset.

In [219]:
df.describe()

comment_id
count  34768.000000
mean      17.651461
std      116.955921
min     -153.000000
25%        1.000000
50%        3.000000
75%       10.000000
max     9449.000000

In [220]:
df.to_csv('processed.csv',index=False,header=True)

In [221]:
processed_df = pd.read_csv('processed.csv')
processed_df = processed_df[['comment_body', 'post_id', 'comment_id']]

In [222]:
print(processed_df.head())

                                        comment_body              post_id  \
0  This post appears to be about vaccines, please...  2020-12-15 14:21:21   
1  Who's ahead in the pool for the third vaccine ...  2020-12-15 14:28:43   
2  "asymptomatic infection was reduced by 63 perc...  2020-12-15 14:42:16   
3  > and appears to prevent the spread of the vir...  2020-12-15 15:36:09   
4  The fact that there are multiple companies mak...  2020-12-15 17:01:14   

   comment_id  
0           1  
1        1784  
2        1905  
3        3503  
4         326  


In [223]:
import nltk
import ssl

try:
    _create_unverified_https_context = ssl._create_unverified_context
except AttributeError:
    pass
else:
    ssl._create_default_https_context = _create_unverified_https_context

nltk.download('stopwords')
nltk.download('punkt')
nltk.download('wordnet')

from nltk.corpus import stopwords
stop_words = stopwords.words('english')
stop_words.extend(['would', 'uhmm'])

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\Person\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\Person\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\Person\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!


In [224]:
# Cleaning Function
def cleaning(string):
    string = string.lower()
    string = string.replace('àö', '')
    string = string.replace('çå', '')
    string = string.replace('ç', '')
    string = string.replace('å', '')
    string = string.replace('äôs', '')
    string = string.replace('çç', '')
    string = string.replace('äôt', '')
    string = string.lower()
   
    from nltk.stem import WordNetLemmatizer
    import nltk
    words = nltk.word_tokenize(string)
    lemmatizer = WordNetLemmatizer()
    words = [lemmatizer.lemmatize(i) for i in words]
    new_words = []
    for i in words:
        if i.isdigit():
            new_words.append('NUMBER')
        else:
            new_words.append(i)
    return ' '.join(new_words)

In [225]:
#df_threads_text = processed_df.groupby('comment_id')['comment_body'].apply(set).reset_index(name='comment_id')
processed_df['comment_body']=processed_df['comment_body'].apply(cleaning)
print(processed_df[['comment_id','comment_body']])

       comment_id                                       comment_body
0               1  this post appears to be about vaccine , please...
1            1784  who 's ahead in the pool for the third vaccine...
2            1905  `` asymptomatic infection wa reduced by NUMBER...
3            3503  > and appears to prevent the spread of the vir...
4             326  the fact that there are multiple company makin...
...           ...                                                ...
34763           2  > if a state receives lower quantity than they...
34764           4  military , intelligence and/or diplomatic barg...
34765           1  or just not evenly allocated to state . again ...
34766           1        i heard alien just beamed it up . zzzzzooop
34767           3  i 'm not sure what you 're trying to argue . t...

[34768 rows x 2 columns]
